app/assets/javascripts/modules/remove-filter.js

```diff
@@ -39,17 +39,9 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
         $input.trigger(onChangeSuppressAnalytics);
       }
       else if (inputType == 'text' || inputType == 'search') {
-        var needle = decodeEntities(removeFilterValue.toString());
-        var splitKeywords = currentVal.split(" ");
-
-        for (var i = 0; i < splitKeywords.length; i++) {
-          if (splitKeywords[i].toString() === needle) {
-            splitKeywords.splice(i, 1);
-            break;
-          }
-        }
-
-        var newVal = splitKeywords.join(" ").trim();
+        var haystack = ' ' + currentVal.trim() + ' ';
+        var needle = ' ' + decodeEntities(removeFilterValue.toString()) + ' ';
+        var newVal = haystack.replace(needle, ' ').replace(/  /g, ' ').trim();
```

Comment:
Nice work here. It may not be clear in the future why this is necessary, so perhaps putting this into a named method  and then calling it here would help? Or perhaps adding a comment / longer commit message giving some context?